lib/Dialect/SPIRV/Serialization/Serializer.cpp

```diff
@@ -436,7 +461,15 @@ LogicalResult Serializer::processType(Location loc, Type type,
                                      operands))) ||
       succeeded(prepareBasicType(loc, type, typeEnum, operands))) {
     typeIDMap[type] = typeID;
-    return encodeInstructionInto(typesGlobalValues, typeEnum, operands);
+
+    if (failed(encodeInstructionInto(typesGlobalValues, typeEnum, operands))) {
+      return failure();
+    }
+
+    if (auto dType = type.dyn_cast<spirv::ArrayType>()) {
```

Comment:
Why is this here? Shouldn't this be within `prepareBasicType` where it is checking for handling ArrayType?

Reply:
As far as I understood, please fix me if I'm wrong, to encode instruction with OpDecorate we need to call `encodeInstructionInto() `with last argument as array of args, when the first element of arg is unique id, so to process it in the prepareBasic function, we have to change the API prepareBasicType, so, if it's ok, I'll change it.
Thanks.

Reply:
Oh, I see the function prepareBasicType is private, so, if it's ok to I'll add one more args as resultId.

Reply:
This looks good. Thanks!